WordPress/Classes/Utility/Media/ItemProviderMediaExporter.swift

```diff
@@ -0,0 +1,117 @@
+import Foundation
+import PhotosUI
+
+/// Manages export of media assets: images and video.
+final class ItemProviderMediaExporter: MediaExporter {
+    var mediaDirectoryType: MediaDirectory = .uploads
+    var imageOptions: MediaImageExporter.Options?
+    var videoOptions: MediaVideoExporter.Options?
+
+    private let provider: NSItemProvider
+
+    init(provider: NSItemProvider) {
+        self.provider = provider
+    }
+
+    func export(onCompletion: @escaping (MediaExport) -> Void, onError: @escaping (MediaExportError) -> Void) -> Progress {
```

Comment:
It works nearly identical to the existing `MediaAssetExporter` (`PHAsset` support) and shares some of the code with it. 

Great post on `NSItemProvider`: https://www.humancode.us/2023/07/08/all-about-nsitemprovider.html.